src/core/Header/HeaderButton.test.tsx

```diff
@@ -55,7 +55,42 @@ it('render isActive correctly', () => {
   expect(activeButton?.getAttribute('aria-current')).toEqual('location');
 });
 
-it('render startIcon correctly', () => {
+it('should render split button correctly', () => {
+  const itemOneOnClick = () => {};
+  const buttonOnClick = () => {};
+
+  const { container } = render(
+    <HeaderButton
+      name={'MockName'}
+      onClick={buttonOnClick}
+      menuItems={(close) => [
+        <MenuItem
+          key={0}
+          onClick={() => {
+            itemOneOnClick();
+            close();
+          }}
+        >
+          Test0
+        </MenuItem>,
+        <MenuItem key={1} onClick={close}>
+          Test1
+        </MenuItem>,
+        <MenuItem key={2} onClick={close}>
+          Test2
+        </MenuItem>,
+      ]}
+    />,
+  );
+
+  const splitButton = container.querySelector('.iui-header-split-button');
+  expect(splitButton).toBeTruthy();
+
+  const innerButtons = splitButton?.querySelectorAll('.iui-borderless');
+  expect(innerButtons?.length).toBe(2);
```

Comment:
I would like to also check that clicking first menu `buttonOnClick` is called, clicking arrow button opens the menu like in the test `should render menuItems correctly`.